tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -0,0 +1,223 @@
+                                                  'PredefinedMetricSpecification': {
+                                                      'PredefinedMetricType': 'DynamoDBWriteCapacityUtilization'
+                                                  },
+                                                  'ScaleOutCooldown': scale_out_cooldown_in_seconds,
+                                                  'ScaleInCooldown': scale_in_cooldown_in_seconds
+                                              })
+
+    def build_key(self, type, iden) -> str:
+        """
+        It builds a unique partition key. The key could be objects with __str__ method.
+        """
+        return f"{type} {iden}"
+
+    def restore(self, keys) -> dict:
+        """
+        Fetch all under the same parition key(keys).
+        ret: <dict of key to states>
+        """
+        # try:
+        items = zip(
+            keys,
+            (self[key] for key in keys),
+        )
+        # except Exception as e:
+        #     self.alert(str(e))
+        return {k: v for k, v in items if v}
+
+    def alert(self, msg: str):
+        import pysensu_yelp
+        result_dict = {
+            'name': 'tron_dynamodb_check',
+            'runbook': '',
+            'status': 1,
+            'output': msg,
+            'team': 'compute-infra',
+            'tip': '',
+            'page': None,
+            'notification_email': 'mingqiz@yelp.com',
+            'irc_channels': None,
+            'slack_channels': None,
+            'alert_after': '1m',
+            'check_every': '10m',
+            'realert_every': -1,
+            'ttl': None
+        }
+        pysensu_yelp.send_event(**result_dict)
+
+    def __getitem__(self, key: ShelveKey) -> object:
+        """
+        It returns an object which is deserialized from binary
+        """
+        table_name = self.name.replace('/', '-')
+        keys = [{'key': {'S': str(key)}, 'index': {'N': str(index)}} for index in range(self._get_num_of_partitions(key))]
+        if not keys:
+            return None
+        vals = self.client.batch_get_item(
+            RequestItems={
+                table_name: {
+                    'Keys': keys,
+                    'ConsistentRead': True
+                },
+            }
+        )['Responses'][table_name]
+        vals.sort(key=lambda x: x['index']['N'])
+        res = bytearray()
+        for val in vals:
+            res += bytes(val['val']['B'])
+
+        return pickle.loads(res) if res else None
+
+    def save(self, key_value_pairs) -> None:
+        """
+        Remove all previous data with the same partition key, examine the size of state_data,
+        and splice it into different parts under 400KB with different sort keys,
+        and save them under the same partition key built.
+        """
+        # try:
+        for key, val in key_value_pairs:
+            self._delete_item(key)
+            self[key] = pickle.dumps(val)
+        # except Exception as e:
```

Comment:
Maybe you should log the key values and exception, at least.

Reply:
Oh right. I forgot that.

Reply:
Okay, works for now.